src/tools/pkcs11-tool.c

```diff
@@ -2830,7 +2833,7 @@ static void decrypt_data(CK_SLOT_ID slot, CK_SESSION_HANDLE session,
 	/* set "default" MGF and hash algorithms. We can overwrite MGF later */
 	switch (opt_mechanism) {
 	case CKM_RSA_PKCS_OAEP:
-		build_rsa_oaep_params(&oaep_params, &mech, NULL, 0);
+		build_rsa_oaep_params(&params.oaep, &mech, NULL, 0);
```

Comment:
As this is growing to more mechanisms, I am wondering what are the difference in encryption and decryption case so it could be all moved into separate function `build_mechanism` or something to avoid code duplication.

Reply:
This PR and the Chacha20 PRs are conflicting.
I propose to add the `build_mechanism` function in the Chacha20 PR and once it is merged, I will rework this one.
Is it OK for you?